backend/inspirehep/records/serializers/bibtex.py

```diff
@@ -16,6 +19,27 @@
 
 
 class BibtexWriter(Writer):
+    def _encode(self, text):
+        import latexcodec  # NOQA
+
+        if "$" not in text:
+            return codecs.encode(text, "ulatex+{}".format(self.encoding))
+        else:
+            math_expression_regex = re.compile(r"(?<!\\)\$.*?\$")
+            found_math_expressions = re.findall(math_expression_regex, text)
+            encoded_text = codecs.encode(text, "ulatex+{}".format(self.encoding))
+            encoded_math_expressions = re.findall(math_expression_regex, encoded_text)
+            assert len(found_math_expressions) == len(
+                encoded_math_expressions
+            ), f"number of math parts in LaTeX-encoded string is different for '{text}'."
+            for encoded_math_expression, found_math_expression in zip(
```

Comment:
again, variable names are a bit long here I think. What about
```suggestion
            for encoded_math, original_math in zip(
```